File: cpp/40_jeff_and_digits.cpp
#include <bits/stdc++.h>
using namespace std;
int main()
{
    int n;
    string s;
    cin >> n >> s;
    int cnt5 = 0, cnt0 = 0;
    for (char c : s)
    {
        if (c == '5')
            cnt5++;
        else if (c == '0')
            cnt0++;
    }
    if (cnt0 == 0)
    {
        cout << -1 << "\n";
        return 0;
    }
    int use5 = cnt5 - cnt5 % 9;
    if (use5 == 0)
    {
        cout << 0 << "\n";
        return 0;
    }
    for (int i = 0; i < use5; i++)
        cout << '5';
    for (int i = 0; i < cnt0; i++)
        cout << '0';
    cout << "\n";
}

File: python/40_jeff_and_digits.py
n = int(input().strip())
s = input().strip()
cnt5 = s.count("5")
cnt0 = s.count("0")
if cnt0 == 0:
    print(-1)
else:
    use5 = cnt5 - (cnt5 % 9)
    if use5 == 0:
        print(0)
    else:
        print("5" * use5 + "0" * cnt0)
